streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -1103,26 +1108,28 @@ public <KG, VG, VR> KStream<K, VR> leftJoin(final GlobalKTable<KG, VG> globalTab
         return globalTableJoin(globalTable, keySelector, joiner, true, named);
     }
 
-    private <KG, VG, VR> KStream<K, VR> globalTableJoin(final GlobalKTable<KG, VG> globalTable,
-                                                        final KeyValueMapper<? super K, ? super V, ? extends KG> keySelector,
-                                                        final ValueJoinerWithKey<? super K, ? super V, ? super VG, ? extends VR> joiner,
-                                                        final boolean leftJoin,
-                                                        final Named named) {
+    private <KGlobalTable, VGlobalTable, VOut> KStream<K, VOut> globalTableJoin(
```

Comment:
nit would `GlobalTableK` be more descriptive than `KGlobalTable` ? same for `VGlobalTable`

Reply:
Guess it's the same question as above -- we might want to use some different name to begin with. Any ideas?

Reply:
Updating to `GlobalKey` and `GlobalValue` for now. Happy to pick something else if you have a good idea.